### ISSUE TITLE
http_proxy environment variable not working

### ISSUE THREAD

**Issue by TLadd** [loadimpact/k6#797]
I am trying to test a proxy server. The [same issue](https://github.com/loadimpact/k6/issues/220) was filed a long while ago and was potentially fixed, but it is not working for me now. My proxy server is running on localhost:3000, and the server being proxied to is on localhost:3001. The requests from k6 do not go through the proxy server and instead go directly to localhost:3000.

I am running the following k6test.js file
```
import http from "k6/http";
import { check } from "k6";

export default function() {
  let res = http.get("http://127.0.0.1:3001/ab-testing", {
    headers: {
      "x-sp-script-id": "ab-testing.js"
    }
  });
  check(res, {
    success: r => r.status == 200
  });
}
```

```
$ export HTTP_PROXY=http://localhost:3000
$ k6 run --http-debug k6test.js

          /\      |‾‾|  /‾‾/  /‾/   
     /\  /  \     |  |_/  /  / /   
    /  \/    \    |      |  /  ‾‾\  
   /          \   |  |‾\  \ | (_) | 
  / __________ \  |__|  \__\ \___/ .io

  execution: local
     output: -
     script: k6test.js

    duration: -,  iterations: 1
         vus: 1, max: 1

Request: [----------------------------------------------------------] starting
GET /ab-testing HTTP/1.1
Host: 127.0.0.1:3001
User-Agent: k6/0.22.1 (https://k6.io/);
X-Sp-Script-Id: ab-testing.js
Accept-Encoding: gzip


Response:
HTTP/1.1 404 Not Found
Content-Length: 149
Connection: keep-alive
Content-Security-Policy: default-src 'self'
Content-Type: text/html; charset=utf-8
Date: Sat, 06 Oct 2018 23:36:39 GMT
X-Content-Type-Options: nosniff
X-Powered-By: Express


    done [==========================================================] 1 / 1

    ✗ success
     ↳  0% — ✓ 0 / ✗ 1

    checks.....................: 0.00% ✓ 0   ✗ 1  
    data_received..............: 398 B 9.2 EB/s
    data_sent..................: 122 B 9.2 EB/s
    http_req_blocked...........: avg=250µs  min=250µs  med=250µs  max=250µs  p(90)=250µs  p(95)=250µs 
    http_req_connecting........: avg=189µs  min=189µs  med=189µs  max=189µs  p(90)=189µs  p(95)=189µs 
    http_req_duration..........: avg=430µs  min=430µs  med=430µs  max=430µs  p(90)=430µs  p(95)=430µs 
    http_req_receiving.........: avg=90µs   min=90µs   med=90µs   max=90µs   p(90)=90µs   p(95)=90µs  
    http_req_sending...........: avg=43µs   min=43µs   med=43µs   max=43µs   p(90)=43µs   p(95)=43µs  
    http_req_tls_handshaking...: avg=0s     min=0s     med=0s     max=0s     p(90)=0s     p(95)=0s    
    http_req_waiting...........: avg=297µs  min=297µs  med=297µs  max=297µs  p(90)=297µs  p(95)=297µs 
    http_reqs..................: 1     +Inf/s
    iteration_duration.........: avg=1.11ms min=1.11ms med=1.11ms max=1.11ms p(90)=1.11ms p(95)=1.11ms
    iterations.................: 1     +Inf/s
    vus........................: 1     min=1 max=1
    vus_max....................: 1     min=1 max=1
```

The equivalent curl command succeeds:

```
$ env HTTP_PROXY="127.0.0.1:3000" curl -H "x-sp-script-id: ab-testing.js" http://127.0.0.1:3001/ab-testing
{"name":"test"}
```

**Comment by TLadd** [comment]
Did a little research and noticed that Go's [ProxyFromEnvironment](https://golang.org/pkg/net/http/#ProxyFromEnvironment) has a special case for localhost. Created an alias in /etc/hosts and it works fine.

**Comment by pawansinha0612** [comment]
@TLadd - I am exactly having the same issue, If this is working for you guys. can you please help me with the code. I have my k6 scripts all in JS, now where should i incorporate ProxyFromEnvironment. please help.

**Comment by TLadd** [comment]
@pawansinha0612 It's been a long while and I don't even have access to the codebase I was using this in anymore. From what I can piece together from what I wrote here though, I think my solution was to no longer specify 127.0.0.1 or localhost for the HTTP_PROXY. Instead, I specified HTTP_PROXY="localtest.me:3000" (localtest.me is one of those special domains that points to localhost) or something else if you use /etc/hosts to map a custom domain to localhost.

The `ProxyFromEnvironment` thing comes into play because k6 is written in Go and uses `ProxyFromEnvironment` for this proxy functionality.

Hope this helps!
